flink-runtime/src/test/java/org/apache/flink/runtime/io/network/partition/hybrid/tiered/tier/memory/MemoryTierProducerAgentTest.java

```diff
@@ -0,0 +1,127 @@
+ */
+
+package org.apache.flink.runtime.io.network.partition.hybrid.tiered.tier.memory;
+
+import org.apache.flink.runtime.io.network.buffer.BufferBuilderTestUtils;
+import org.apache.flink.runtime.io.network.partition.ResultPartitionID;
+import org.apache.flink.runtime.io.network.partition.hybrid.tiered.TestingTieredStorageMemoryManager;
+import org.apache.flink.runtime.io.network.partition.hybrid.tiered.common.TieredStorageIdMappingUtils;
+import org.apache.flink.runtime.io.network.partition.hybrid.tiered.common.TieredStoragePartitionId;
+import org.apache.flink.runtime.io.network.partition.hybrid.tiered.common.TieredStorageSubpartitionId;
+import org.apache.flink.runtime.io.network.partition.hybrid.tiered.netty.TestingNettyConnectionWriter;
+import org.apache.flink.runtime.io.network.partition.hybrid.tiered.netty.TestingNettyServiceProducer;
+import org.apache.flink.runtime.io.network.partition.hybrid.tiered.netty.TestingTieredStorageNettyService;
+import org.apache.flink.runtime.io.network.partition.hybrid.tiered.storage.TieredStorageResourceRegistry;
+
+import org.junit.jupiter.api.Test;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+/** Tests for {@link MemoryTierProducerAgent}. */
+class MemoryTierProducerAgentTest {
+
+    private static final int NUM_SUBPARTITIONS = 10;
+
+    private static final int BUFFER_SIZE = 1024;
+
+    private static final int SEGMENT_SIZE_BYTES = 1024;
+
+    private static final TieredStoragePartitionId PARTITION_ID =
+            TieredStorageIdMappingUtils.convertId(new ResultPartitionID());
+    private static final TieredStorageSubpartitionId SUBPARTITION_ID =
+            new TieredStorageSubpartitionId(0);
+
+    @Test
+    void testTryStartNewSegment() {
+        MemoryTierProducerAgent memoryTierProducerAgent = createMemoryTierProducerAgent(false);
+        assertThat(memoryTierProducerAgent.tryStartNewSegment(SUBPARTITION_ID, 0)).isFalse();
+        memoryTierProducerAgent.connectionEstablished(
+                SUBPARTITION_ID, new TestingNettyConnectionWriter.Builder().build());
+        assertThat(memoryTierProducerAgent.tryStartNewSegment(SUBPARTITION_ID, 0)).isTrue();
+    }
+
+    @Test
+    void testTryWrite() {
+        MemoryTierProducerAgent memoryTierProducerAgent = createMemoryTierProducerAgent(false);
+        memoryTierProducerAgent.connectionEstablished(
+                SUBPARTITION_ID, new TestingNettyConnectionWriter.Builder().build());
+        assertThat(
+                        memoryTierProducerAgent.tryWrite(
+                                SUBPARTITION_ID, BufferBuilderTestUtils.buildSomeBuffer()))
+                .isTrue();
+        assertThat(
+                        memoryTierProducerAgent.tryWrite(
+                                SUBPARTITION_ID, BufferBuilderTestUtils.buildSomeBuffer()))
+                .isFalse();
```

Comment:
This strongly rely on `BUFFER_SIZE = SEGMENT_SIZE_BYTES = 1024`, this will make this test case a bit obscure. I suggest setting the default `SEGMENT_SIZE_BYTES` in the class to be greater than `BUFFER_SIZE`(e.g. `BUFFER_SIZE * 2`) and introducing `createMemoryTierProducerAgent (boolean isBroadcast, int segmentSizeBytes)`. As for this test, we can call `createMemoryTierProducerAgent(false, BUFFER_SIZE)`